kernel/src/proc/process.c

```
@@ -319,10 +386,141 @@ void process_cleanup_child(pcb_t *child)
+    child->regs = parent->regs;
+    child->regs.esp = (uint32_t)tf;  /* point to child's own trap frame */
+    child->regs.eax = 0;
+    child->regs.cr3 = (uint32_t)child_dir;  /* restore CR3 overwritten by struct copy */
+
+    /* Clone VMA list (deep copy) */
+    child->memory_regions = NULL;
+    vma_t *src_vma = parent->memory_regions;
+    vma_t *dst_vma_last = NULL;
+    while (src_vma) {
+        vma_t *new_vma = malloc(sizeof(vma_t));
+        new_vma->start = src_vma->start;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)